### PR TITLE
Fixes some Ruby linter errors

### PR DESCRIPTION
**Pull request by rossjones**
The govuk linter complains a lot, and this PR will make it complain
a little less. The fixes are mostly just whitespace changes though 
there are a few actual changes (block.call -> yield for instance).

### CODE REVIEW

File: spec/features/viewing_metrics_spec.rb [hunk 1]
```diff
@@ -25,10 +25,10 @@
         all('a', text: /\AOpen\z/).each(&:click) if javascript_enabled
 
         expect(metric_groups(:name, :transactions_received_total)).to eq([
-          ["Total for UK government",                               "746067381"],
```
Comment (olly) [inline spec/features/viewing_metrics_spec.rb:28]: This is weird, it's removed the whitespace for some lines and not others. It should probably be consistent.

File: spec/support/metric_group_header_shared_examples.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ def render
     super(partial: partial, locals: { metric_group: metric_group })
   end
 
-  it "has a 'm-metric-group-header' container"  do |example|
+  it "has a 'm-metric-group-header' container" do |_example|
```
Comment (olly) [inline spec/support/metric_group_header_shared_examples.rb]: I think this is extraneous and could just be removed.

**Review by olly — COMMENTED** [review]
I'm all in favour of linters, we just never got round to setting it up.

I think we should probably add this to CI too, for continuous enforcement.

**Comment by rossjones** [comment]
The underscores were done manually because the style insists on normalcase (no _ or camels allowed).  It looks pretty horrible.  How about I go with metric_group_one instead?

**Pull request merged**